tests/vmipreset_test.go

```diff
@@ -170,10 +170,12 @@ var _ = Describe("VMIPreset", func() {
 
 			newPreset := waitForPreset(virtClient, memoryPrefix)
 
-			err = virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(tests.NamespaceTestDefault).Body(vmi).Do().Error()
+			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
 			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
 
-			newVMI := waitForVirtualMachine(virtClient)
+			newVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &k8smetav1.GetOptions{})
```

Comment:
like the other test I pointed out, you shouldn't need a GET here. The newVMI can be returned in the Create with all the changes... or at least that's supposed to be the way it works :D